agent-service/shared/llm/embeddings_factory.py

```
@@ -23,6 +22,13 @@ def create_embeddings(
         model_name = model_params.get("local_embedding_model")
         if not model_name:
             raise ValueError("local_embedding_model must be set for local embeddings")
+        try:
+            from langchain_huggingface import HuggingFaceEmbeddings
+        except ImportError:
+            raise ImportError(
+                "langchain-huggingface is required for local embeddings. "
+                "Install it with: poetry install --with local-models"
+            )
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)